ui/component/transactionListTable/internal/txo-list-item.jsx

```diff
@@ -0,0 +1,152 @@
+// @flow
+import * as ICONS from 'constants/icons';
+import React from 'react';
+import ButtonTransaction from 'component/common/transaction-link';
+import CreditAmount from 'component/common/credit-amount';
+import DateTime from 'component/dateTime';
+import Button from 'component/button';
+import Spinner from 'component/spinner';
+import { buildURI, parseURI, TXO_LIST as TXO, TXO_ABANDON_STATES as TXO_STATES } from 'lbry-redux';
+
+type Props = {
+  txo: Txo,
+  revokeClaim: (Txo, (string) => void) => void,
+  isRevokeable: boolean,
+  reward: ?{
+    reward_title: string,
+  },
+};
+
+type State = {
+  abandonState: string,
+};
+
+class TxoListItem extends React.PureComponent<Props, State> {
+  constructor() {
+    super();
+    this.state = { abandonState: TXO_STATES.READY };
+    (this: any).abandonClaim = this.abandonClaim.bind(this);
+    (this: any).getLink = this.getLink.bind(this);
+  }
+
+  getLink(type: string) {
+    const { abandonState } = this.state;
+    if (this.state.abandonState === TXO_STATES.PENDING) {
+      return <Spinner type={'small'} />;
+    }
+    if (type === TXO.SUPPORT) {
+      return (
+        <Button
+          disabled={abandonState === TXO_STATES.DONE || abandonState === TXO_STATES.ERROR}
+          button="secondary"
+          icon={ICONS.UNLOCK}
+          onClick={this.abandonClaim}
+          title={__('Unlock Tip')}
+        />
+      );
+    }
+    const abandonTitle = type === TXO.SUPPORT ? 'Abandon Support' : 'Abandon Claim';
+    return (
+      <Button
+        disabled={abandonState === TXO_STATES.DONE || abandonState === TXO_STATES.ERROR}
+        button="secondary"
+        icon={ICONS.DELETE}
+        onClick={this.abandonClaim}
+        title={__(abandonTitle)}
+      />
+    );
+  }
+
+  abandonClaim() {
+    this.props.revokeClaim(this.props.txo, st => this.setState({ abandonState: st }));
```

Comment:
I would rename `st` to something more meaningful. You could just call it `abandonState`